src/renovate.ts

```diff
@@ -41,6 +49,23 @@ class Renovate {
     }
   }
 
+  /**
+   * Fetch the host docker group of the GitHub Action runner.
+   *
+   * The Renovate container needs access to this group in order to have the
+   * required permissions on the Docker socket.
+   */
+  private getDockerGroupId(): string {
+    const groupInfo = execSync(`getent group docker`)
```

Comment:
why do you changed this? i don't like the exec here. would prefer to use `/etc/groups` again.

Reply:
I'm wondering why you don't like exec here, I changed it because I see no point in reimplementing a file parsing that is already done by a UNIX binary, this GitHub action is literally a bash script with a JavasScript mustache: you use exec to do `docker run` which is the whole point of this GitHub action.

I'm not exactly sure what you wanted when you asked "use static regex and use named capture group", do you want to add a Regex snippet since the previous one was not satisfactory?

Reply:
i don't like sync exec. I also plan to support #571 in future.

Reply:
i said before to just refactor the regex and not to change the whole function to use other stuff

Reply:
> i don't like sync exec. I also plan to support https://github.com/renovatebot/github-action/issues/571 in future.

well, reading `/etc/groups` on Windows will be as impossible as using `getent` and `getent` is supported on macOS. but nothing like this will work anyway because Docker runs in a VM on macOS/Windows and mapping group/mounting the Docker socket will be a different story.

Reply:
is it satisfactory?
```typescript
const re = new RegExp('^docker:x:(?<groupId>[1-9][0-9]*):', 'm');
    const match = re.exec(groups);
    if (match?.groups?.groupId === undefined) {
      throw new Error(`Could not find group docker in ${groupFile}`);
    }
```